### ISSUE TITLE
Invalid superglobal - request.getPost inside volt file

### ISSUE THREAD

**Issue by hufersil** [phalcon/cphalcon#13252]
Using Phalcon < 3.2 I can use `{{ request.getPost('varname') }}` (inside volt template) without problems.

Once we upgrade to 3.2.4, we get this exception: "Invalid superglobal".
If called inside a controller ( `$this->request->getPost('varname')` ), there is no problem.

### Expected and Actual Behavior
Expected - get post content of varname index or null if not present inside a volt template.
Actual behavior - throwing exception.

> **Describe what you are trying to achieve and what goes wrong.**

Get post var using `request.getPost('varname')`

> Provide output if related. Provide coredump if any. Use https://github.com/phalcon/cphalcon/wiki/Generating-a-backtrace as reference

inside a volt template:
```php
{{ request.getPost('varname') }}
```

* Phalcon version: 3.2.4
* PHP Version: 7.0.22-0ubuntu0.16.04.1
* Operating System: Linux ip-172-31-44-144 4.4.0-87-generic #110-Ubuntu SMP Tue Jul 18 12:55:35 UTC 2017 x86_64 
* Server: Nginx

**Comment by Jurigag** [comment]
Does $_POST return anything? This error is happening when superglobal doesn't exist like you have enable_post_data_reading set to disabled.

**Comment by hufersil** [comment]
but why does this work inside a controller, even in a get request?

and also as mentioned before, this works on phalcon < 3.2.


Em 13 de jan de 2018 15:20, "Wojciech Ślawski" <notifications@github.com>
escreveu:

Does $_POST return anything? This error is happening when superglobal
doesn't exist like you have enable_post_data_reading set to disabled.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/phalcon/cphalcon/issues/13252#issuecomment-357451155>,
or mute the thread

**Comment by Jurigag** [comment]
Hmmm it might be about symbol tables. Just pass this value from controller to view as workaround. There was update to zephir which added check if superglobal is valid. I will check it next week if there is indeed any probleem.

**Comment by hufersil** [comment]
yeah... this is what I'm doing for now.

at localhost I'm using phalcon 3.0 and works just fine.

at prod with 3.2 the error occurs.

tks

Em 13 de jan de 2018 15:54, "Wojciech Ślawski" <notifications@github.com>
escreveu:

Hmmm it might be about symbol tables. Just pass this value from controller
to view as workaround. There was update to zephir which added check if
superglobal is valid. I will check it next week if there is indeed any
probleem.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/phalcon/cphalcon/issues/13252#issuecomment-357453566>,
or mute the thread

**Comment by hufersil** [comment]
Guys I don't know what happend with this update, but there are others place that was working and now it isn't...

This code that is working correctly with phalcon < 3.2:
```php
{{ select(item.name, item.options, 'label': item.label, 'using': [item.optionKey, item.optionLabel], 'simple':true, 'emptyText': item.label) }}
```

Throws this exception with 3.2:

Invalid superglobal#0 [internal function]: Phalcon\Tag::getValue('coddificuldade', Array)
#1 [internal function]: Phalcon\Tag\Select::selectField(Array, NULL)

**Comment by mruz** [comment]
I can confirm the issue, @sergeyklay any update?

**Comment by sergeyklay** [comment]
Let's use https://github.com/phalcon/zephir/issues/1623 to discuss this issue

**Comment by sergeyklay** [comment]
Fixed in the `3.4.x` branch. Feel free to open a new issue if the problem appears again. Thank you for contributing.
